dockers/docker-dhcp-server/dhcpservd.py

```diff
@@ -0,0 +1,521 @@
+                    for port_name, ip_ranges in ports.items():
+                        if port_name in classes_port:
+                            class_set.append(client_class[port_name])
+                        for ip_range in ip_ranges:
+                            pool = {
+                                "pool": "{} - {}".format(str(ip_range[0]), str(ip_range[1])),
+                                "client-class": "{}:{}".format(hostname, port_name)
+                            }
+                            config_subnet["pools"].append(pool)
+                    # TODO Add customized options
+                    self.kea_config["Dhcp4"]["subnet4"].append(config_subnet)
+        if len(class_set) != 0:
+            self.kea_config["Dhcp4"]["client-classes"] = class_set
+
+        with open("/etc/kea/kea-dhcp4.conf", "w") as file:
+            json.dump(self.kea_config, file, indent=4, ensure_ascii=False)
+
+    def clear_table(self, db, table_name):
+        """
+        Clear table in db.
+        Args:
+            db: Db object.
+            table_name: Name of table to be clear
+        """
+        for key in db.keys("{}*".format(table_name)):
+            db.delete(key)
+
+    def update_lease_handler(self, signum, frame):
+        """
+        Hanlder function to update lease table in STATE_DB
+        """
+        # Read lease file generated by kea-dhcp4
+        lease_obj = {}
+        with open(LEASE_FILE_PATH, "r", encoding="utf-8") as fb:
+            dq = deque(fb)
+        while dq:
+            last_row = dq.pop()
+            splits = last_row.split(",")
+            if splits[0] == "address":
+                break
+            ip_str = splits[0]
+            mac_address = splits[1]
+            if mac_address in lease_obj:
+                continue
+            valid_lifetime = splits[3]
+            lease_end = splits[4]
+            lease_obj[mac_address] = {
+                "lease_start": str(int(lease_end) - int(valid_lifetime)),
+                "lease_end": lease_end,
+                "ip": ip_str
+            }
+        # Clear lease table in STATE_DB
+        self.clear_table(self.state_db, DHCP_SERVER_IPV4_LEASE)
+        # Update lease table
+        for mac_address, value in lease_obj.items():
+            ip_str = value["ip"]
+            if value["lease_start"] == value["lease_end"]:
+                continue
+            for net in self.ip_ports.keys():
+                if self.ip_in_net(ipaddress.ip_address(ip_str), net):
+                    key = "{}|{}|{}".format(DHCP_SERVER_IPV4_LEASE, self.ip_ports[net], mac_address)
+                    for k, v in value.items():
+                        self.state_db.hset(key, k, v)
+
+    def add_signal_handler(self):
+        """
+        Add signal handler
+        """
+        # Listen SIGUSR1 to update lease table.
+        signal.signal(signal.SIGUSR1, self.update_lease_handler)
+
+    def sighup_process(self, process_name):
```

Comment:
The only usage of this method is with input "kea-dhcp4", why create a method? 
Can we have a class member such as `self.dhcp_server_process_name` and then you can modify the method signature to `def notify_dhcp_server_process(self)`